bob-access/src/credentials.rs

```diff
@@ -59,13 +112,36 @@ impl CredentialsKind {
     }
 }
 
-impl Credentials {
-    pub fn builder() -> CredentialsBuilder {
-        CredentialsBuilder::default()
+impl DeclaredCredentials {
+    pub fn builder() -> DeclaredCredentialsBuilder {
+        DeclaredCredentialsBuilder::default()
+    }
+
+    pub fn ip(&self) -> &Option<Vec<SocketAddr>> {
+        &self.address
+    }
+
+    pub fn add_addresses(&mut self, mut addresses: Vec<SocketAddr>) {
```

Comment:
1. It is a bit strange to pass `addresses` vector with `mut` modifier.
2. Why it is `add_addresses`? If resolved addresses changed, you should replace old ones with new (whole collection). Otherwise it is possible to get duplicates.